### PR TITLE
feat: repositoryUrls

### LINKED ISSUE #2181: How do I add support for different pypi providers?
#### This is a:

- [ ] Bug report (non-security related)
- [ ] Feature request
- [x] I'm not sure which of those it is

#### I'm using:

- [x] The Renovate GitHub App
- [ ] Self-hosted GitHub
- [ ] Self-hosted GitLab
- [ ] Self-hosted VSTS

#### Please describe the issue:

I see that [public pypi](https://github.com/renovatebot/renovate/blob/master/lib/datasource/pypi.js#L12) is the sensible default. I'd like to make it so that either `PIP_INDEX_URL` from the environment or `--pip-index-url` from the command line would override the host, allowing for private/other hosts.

Where should I get started in writing the test for that?

### PR DESCRIPTION
**Pull request by rarkins**
Adds config option repositoryUrls which can be used by pip to define an alternate host to pypi.

Closes #2181

### CODE REVIEW

**Comment by rarkins** [comment]
@khornberg please review

**Comment by rarkins** [comment]
Does not include `pip.conf` support yet, as I'm not sure if there's anywhere else we should logically look for the `pip.conf` except for `~/pip.conf`

File: lib/manager/pip_requirements/extract.js [hunk 1]
```diff
@@ -16,6 +16,13 @@ module.exports = {
 function extractDependencies(content) {
   logger.debug('pip_requirements.extractDependencies()');
 
+  let registryUrls;
+  const [firstLine] = content.split('\n');
```
Comment (hbetts) [inline lib/manager/pip_requirements/extract.js]: The `--index-url` option can technically come on any line before the first package is specified.

The following reference shows that options are first in the file, and that options may be given in any order - https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format

File: lib/manager/pip_requirements/extract.js [hunk 2]
```diff
@@ -26,13 +33,18 @@ function extractDependencies(content) {
...
+      const dep = {
         depName,
         currentValue,
         lineNumber,
         purl: 'pkg:pypi/' + depName,
         versionScheme: 'pep440',
+        registryUrls,
       };
+      if (registryUrls) {
+        dep.registryUrls = registryUrls;
```
Comment (hbetts) [inline lib/manager/pip_requirements/extract.js]: This seems redundant with line 42. :thinking:

File: lib/manager/pip_requirements/extract.js (hunk 1 above)
Comment (rarkins) [reply lib/manager/pip_requirements/extract.js]: I've modified it to look on *every* line, as that should catch all valid configurations without being too complex.

**Review by hutson — APPROVED** [review]

**Pull request merged**